plugin-server/src/main/ingestion-queues/analytics-events-ingestion-overflow-consumer.ts

```diff
@@ -49,3 +54,45 @@ export const startAnalyticsEventsIngestionOverflowConsumer = async ({
 
     return queue
 }
+
+export async function eachBatchIngestionFromOverflow(
+    payload: EachBatchPayload,
+    queue: IngestionConsumer
+): Promise<void> {
+    function groupIntoBatchesIngestion(kafkaMessages: KafkaMessage[], batchSize: number): KafkaMessage[][] {
+        // Once we see a distinct ID we've already seen break up the batch
+        const batches = []
+        const seenIds: Set<string> = new Set()
+        let currentBatch: KafkaMessage[] = []
+
+        for (const message of kafkaMessages) {
+            if (currentBatch.length >= batchSize) {
+                seenIds.clear()
+                batches.push(currentBatch)
+                currentBatch = []
+            }
+
+            currentBatch.push(message)
+        }
+
+        if (currentBatch) {
+            batches.push(currentBatch)
+        }
+
+        return batches
+    }
+
+    await eachBatch(payload, queue, eachMessageIngestionFromOverflow, groupIntoBatchesIngestion, 'ingestion')
+}
+
+export async function eachMessageIngestionFromOverflow(message: KafkaMessage, queue: IngestionConsumer): Promise<void> {
+    const pluginEvent = formPipelineEvent(message)
+    // Warnings are limited to 1/key/hour to avoid spamming.
+    if (pluginEvent.team_id && WarningLimiter.consume(`${pluginEvent.team_id}:${pluginEvent.distinct_id}`, 1)) {
+        captureIngestionWarning(queue.pluginsServer.hub.db, pluginEvent.team_id, 'ingestion_capacity_overflow', {
```

Comment:
Are we still capturing this in the main consumer btw?

We'll want to be capturing it there as we want the feedback for this to be fast, if the slow lane is backed up (which is fine and expected) we don't want the warning to be delayed as well.

Reply:
Not sure we care about this being that fast 🤔  besides we're likely sending events there from capture already, so just having it here seems fine 🤷‍♀️

Reply:
The driver here is that we want to avoid users having to ask us where their events are, which will take up our time and theirs, so the faster the feedback the better in this respect.

Fine either way.